tests/test_in_progress_sla_trending_report/test_in_progress_sla_trending_report_outputs.py

```diff
@@ -0,0 +1,2402 @@
+                        sendingSupplierName="EMIS",
+                        requestingSupplierName="TPP",
+                    )
+                ),
+                sourcetype="myevent",
+            )
+
+            # Act
+            test_query = self.generate_splunk_query_from_report(
+                "gp2gp_in_progress_sla_trending_report/gp2gp_in_progress_sla_trending_report_count"
+            )
+
+            test_query = set_variables_on_query(
+                test_query,
+                {
+                    "$index$": index_name,
+                    "$start_time$": report_start.strftime("%Y-%m-%dT%H:%M:%S%z"),
+                    "$end_time$": report_end.strftime("%Y-%m-%dT%H:%M:%S%z"),
+                    "$cutoff$": cutoff,
+                    "$time_period$": "day",
+                },
+            )
+
+            sleep(2)
+
+            telemetry = get_telemetry_from_splunk(
+                self.savedsearch(test_query), self.splunk_service
+            )
+            self.LOG.info(f"telemetry: {telemetry}")
+
+            # Assert
+            time_period_day_today = datetime_utc_now().strftime("%y-%m-%d")
+            time_period_day_today_less_one_day = (datetime_utc_now() - timedelta(days=1)).strftime("%y-%m-%d")
+            time_period_day_today_less_two_day = (datetime_utc_now() - timedelta(days=2)).strftime("%y-%m-%d")
+            time_period_day_today_less_three_day = (datetime_utc_now() - timedelta(days=3)).strftime("%y-%m-%d")
+            expected_values = {
+                "0": {"time_period": time_period_day_today_less_three_day,
+                      "IF": "0",
+                      "B24": "0",
+                      "BEhrS": "0",
+                      "BEhrR": "1"
+                      },
+                "1": {"time_period": time_period_day_today_less_two_day,
+                      "IF": "0",
+                      "B24": "0",
+                      "BEhrS": "1",
+                      "BEhrR": "0"
+                      },
+                "2": {"time_period": time_period_day_today_less_one_day,
+                      "IF": "0",
+                      "B24": "1",
+                      "BEhrS": "0",
+                      "BEhrR": "0"
+                      },
+                "3": {"time_period": time_period_day_today,
+                      "IF": "1",
+                      "B24": "0",
+                      "BEhrS": "0",
+                      "BEhrR": "0"
+                      },
+            }
+
+            for idx in range(len(expected_values)):
+                for key, value in expected_values[str(idx)].items():
+                    self.LOG.info(
+                        f'.[{idx}] | select( ."{key}"=="{value}")'
+                    )
+                    assert jq.first(
+                        f'.[{idx}] | select( ."{key}"=="{value}")',
+                        telemetry,
+                    )
+
+        finally:
+            self.delete_index(index_name)
```

Comment:
There are a lot of lines of code in this. 😵‍💫 
Would it be possible to split the tests into separate files, e.g. a test file for the query it is testing?